src/unit/test_sds.c

```diff
@@ -262,5 +263,97 @@ int test_sds(int argc, char **argv, int flags) {
     TEST_ASSERT_MESSAGE("sdsReszie() crop strlen", strlen(x) == 4);
     TEST_ASSERT_MESSAGE("sdsReszie() crop alloc", sdsalloc(x) >= 4);
     sdsfree(x);
+
+    return 0;
+}
+
+int test_typesAndAllocSize(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    sds x = sdsnewlen(NULL, 31);
+    TEST_ASSERT_MESSAGE("len 31 type", (x[-1] & SDS_TYPE_MASK) == SDS_TYPE_5);
+    sdsfree(x);
+
+    x = sdsnewlen(NULL, 32);
+    TEST_ASSERT_MESSAGE("len 32 type", (x[-1] & SDS_TYPE_MASK) >= SDS_TYPE_8);
+#ifdef USE_JEMALLOC
+    TEST_ASSERT_MESSAGE("len 32 sdsAllocSize", sdsAllocSize(x) == s_malloc_size(sdsAllocPtr(x)));
+#endif
+    sdsfree(x);
+
+    x = sdsnewlen(NULL, 252);
+    TEST_ASSERT_MESSAGE("len 252 type", (x[-1] & SDS_TYPE_MASK) >= SDS_TYPE_8);
+#ifdef USE_JEMALLOC
+    TEST_ASSERT_MESSAGE("len 252 sdsAllocSize", sdsAllocSize(x) == s_malloc_size(sdsAllocPtr(x)));
```

Comment:
I found the warn message from jemalloc manual. 

> Any discrepancy between the requested allocation size and the size reported by malloc_usable_size() should not be depended on, since such behavior is entirely implementation-dependent.

Reply:
We don't depend on discrepancy between the requested allocation size and the size reported by `malloc_usable_size`. Not in the test, not in the code. The test checks that `sdsAllocSize` returns the same value as `s_malloc_size`. It's always true as `sdsAllocSize` is based on the value we got from `malloc_usable_size` before.
In any case, this test checks that our assumption is correct. As we vendor jemalloc, the only case when it can change (theoretically) is when we update jemalloc.

Reply:
> It's always true as sdsAllocSize is based on the value we got from malloc_usable_size before.

You are right, so we can open assertion not only for jemalloc?

Reply:
Oh, you're right! This `ifdef` isn't needed anymore. Updated the PR.